### ISSUE TITLE
Error when using imported string macros after a unit conversion

### ISSUE THREAD

**Issue by imartincei** [Proektsoftbg/Calcpad#387]
![image](https://github.com/user-attachments/assets/ee6b8f59-1b67-48b9-87a4-0d2a4b56afe5)

![image](https://github.com/user-attachments/assets/ec192555-ac46-4c16-84fe-45a0001e348f)

When using a string from an imported macro after a unit conversion, I am getting an error as shown in the bottom photo. The top photo shows the same macro working as intended when it is not imported.

You can also see in these photos that [issue #371](https://github.com/Proektsoftbg/Calcpad/issues/371) is still happening when macros are imported.

**Comment by Proektsoftbg** [comment]
Hi! Maybe you have some additional space character out of the comments that is not seen but causes the trouble. BTW try the simpler single line macro function:
`#def note$(text$) = '<em>&ensp;text$</em>`

Also, remove the internal quotes as shown above and also pass the text without quotes.

**Comment by Proektsoftbg** [comment]
Hi! It can be seen from the pictures that the fix of issue #371 npw works for some lines, and doesn't for others.
This is strange and I cannot figure out why. Can you please post your code, so I can try to reproduce it. On my side it works.

**Comment by imartincei** [comment]
I attached the case showing it working with a macro in the same file but not working when the same function is loaded in from a different file. I tried removing the quotes around text$ and passing the text without quotes and got the same error.

[CalcPad Bug.zip](https://github.com/user-attachments/files/18604688/CalcPad.Bug.zip)

**Comment by Proektsoftbg** [comment]
Hi! Thank you for sending the sourse codes. If you make the macro exaclty as posted above, it always work.
`#def note$(text$) = '<em>&ensp;text$</em>
`
![Image](https://github.com/user-attachments/assets/72624e82-aca7-47fc-8309-275e043c43c0)

But I will check why it doesn't when multiline. Somehow it inserts some unwanted symbol.

**Comment by Proektsoftbg** [comment]
Yes, it inserts the leading tab when the macro is multiline. This is not a problem in other cases, but target units does not allow other code than the units themselves. To avoid problems, always use inline macro definitions for anything that is supposed to be used inline, e.g. on the same line with other things.


However, I also added space trimming around the units to avoid such problems. I will ammend the same version 7.1.7 now but it will be officially available in the next version.

**Comment by imartincei** [comment]
I see, thanks for the clarification, it is working as expected now.
